paasta_tools/kubernetes_tools.py

```diff
@@ -1911,13 +1911,10 @@ def get_node_affinity(self) -> Optional[V1NodeAffinity]:
             required_during_scheduling_ignored_during_execution=selector,
         )
 
-    def get_pod_anti_affinity(self) -> Optional[V1PodAntiAffinity]:
-        """
-        Converts the given anti-affinity on service and instance to pod
-        affinities with the "paasta.yelp.com" prefixed label selector
-        :return:
-        """
-        conditions = self.config_dict.get("anti_affinity", [])
+    def get_pod_anti_affinity_terms(
+        self, config_key: str
```

Comment:
we can probably type config_key as a Literal, but that's probably not worth it atm

Reply:
easy enough, and it caught the fact that I never defined a type in `KubernetesDeploymentConfigDict` for `anti_affinity_preferred`.